pkg/deploy/deploy.go

```diff
@@ -242,19 +242,34 @@ func (d *deployer) removeOldMetricAlerts(ctx context.Context) error {
 // from d.config.Configuration.
 func (d *deployer) getParameters(ps map[string]interface{}) *arm.Parameters {
 	m := map[string]interface{}{}
-
 	v := reflect.ValueOf(*d.config.Configuration)
 	for i := 0; i < v.NumField(); i++ {
-		m[strings.SplitN(v.Type().Field(i).Tag.Get("json"), ",", 2)[0]] = v.Field(i).Interface()
+		if v.Field(i).IsNil() {
```

Comment:
this will panic if someone introduces a non-"pointer" type (i.e. not pointer, map, slice) in the configuration struct.  Either unit test the struct type using reflection or at least add a comment by the type that all fields must be nillable.

Reply:
I recall you @petrkotas added test here, but it is gone again :confused:

Reply:
I do believe every possible scenario is tested `deploy_test.go`. Namely: https://github.com/Azure/ARO-RP/pull/848/files#diff-561b0c09cec330940d2193721c6fe7f6R65

I renamed the test names, and I apologize for that. I know it is not usual and it is not best practice. However, the original test names were not exactly reflecting tested behavior.

I will create separate PR if required. Although, I feel this change is related to my code and it has to be done.

Reply:
@mjudeikis you are right, the test is back where it was. I accidentally removed it when cleaning last commit.